pkg/testing/command/command.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package command provides a generic way for running testing commands.
+package command
+
+import (
+	"bytes"
+	"errors"
+	"os/exec"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type Command struct {
```

Comment:
We will later replace integration logic with these files, while also converting the built-in gadgets test to image-based gadgets (that would be in another PR).

Reply:
The testing package would be used by users with custom gadgets to implement their tests and we would also replace the current integration tests, where instead of using built-in gadgets (most tests use these) we would use the gadget images, some already exist: `run*` tests in `integration/inspektor-gadget`.